File: java/src/main/java/section2/iteration/Iteration.java
package section2.iteration;

import java.util.Arrays;
import java.util.List;

public class Iteration {

    public static void main(String[] args) {

    }

    private void listIteration() {
        List<Long> numbers = Arrays.asList(1L, 2L, 3L);
        for (Long number : numbers) {
            System.out.println(number);
        }
    }

    private void basicFor() {
        for (int i = 0; i < 10; i++) {
            System.out.println(i);
        }
    }
}

File: kotlin/src/main/kotlin/section2/iteration/Iteration.kt
package section2.iteration

fun main() {
}

fun listIteration() {
    val numbers = listOf(1, 2, 3)
    for (number in numbers) {
        println(number)
    }
}

fun basicFor() {
    for (i in 1..10) {
        println(i)
    }
}

fun basicForWithDown() {
    for (i in 10 downTo 1) {
        println(i)
    }
}

fun basicForWithStep() {
    for (i in 1..10 step 2) {
        println(i)
    }
}
